src/lxc/lxc_init.c

```diff
@@ -92,6 +92,86 @@ static struct arguments my_args = {
 	.shortopts = short_options
 };
 
+static void prevent_forking(void)
+{
+	FILE *f;
+	char name[PATH_MAX], path[PATH_MAX];
+	int ret;
+
+	f = fopen("/proc/self/cgroup", "r");
+	if (!f) {
+		SYSERROR("opening /proc/self/cgroup");
+		return;
+	}
+
+	while (!feof(f)) {
+		int fd;
+
+		if (2 != fscanf(f, "%*d:%[^:]:%s", name, path)) {
```

Comment:
This will fail on systems that have the unified hierarchy enabled and where the unified hierarchy entry under `/proc/self/cgroup` comes before the `pids` cgroup entry:

    0::/user.slice/user-1000.slice/session-1.scope